sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_encode.py

```diff
@@ -116,8 +127,8 @@ def encode_ubyte(
     try:
         output.extend(_construct(ConstructorBytes.ubyte, with_constructor))
         output.extend(struct.pack(">B", abs(value)))
-    except struct.error:
-        raise ValueError("Unsigned byte value must be 0-255")
+    except struct.error as exc:
```

Comment:
should these all be from None in _encode.py

Reply:
I did a quick test to see what the `struct.error` here actually looks like:
```
>>> struct.pack(">B", abs(2000))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
struct.error: ubyte format requires 0 <= number <= 255
```
I think this seems reasonable and informative - even if it largely repeats the message we put in a ValueError.
So I think using `from exc` for all these `struct` errors would be appropriate.
I only tend to use `from None` when the underlying error is misleading/unhelpful - like an `AttributeError`.